src/site/content/en/payments/android-payment-apps-delegation/index.md

```diff
@@ -0,0 +1,408 @@
+            } else {
+                service?.changePaymentMethod(mMethodData, mCallback)
+            }
+        } catch (e: RemoteException) {
+            // Handle the remote exception
+        }
+    }
+}
+```
+
+#### `changePaymentMethod`
+
+Notifies the merchant about changes in user selected payment method. The
+`paymentHandlerMethodData` bundle contains `methodName` and optional `details`
+keys both with string values. Chrome will check for a non-empty bundle with a
+non-empty `methodName` and send an `updatePaymentDetails` with one of the
+following error messages via `callback.updateWith` if the validation fails.
+
+```json
+'Method data required.'
+'Method name required.'
+```
+
+#### `changeShippingOption`
+
+Notifies the merchant about changes in user selected shipping option.
+`shippingOptionId` should be the identifier of one of the merchant specified
+shipping options. Chrome will check for a non-empty `shippingOptionId` and send
+an `updatePaymentDetails` with the following error message via
+`callback.updateWith` if the validation fails.
+
+```json
+'Shipping option identifier required.'
+```
+
+#### `changeShippingAddress`
+
+Notifies the merchant about changes in user provided shipping address. Chrome
+will check for a non-empty `shippingAddress` bundle with a valid `countryCode`
+and send an `updatePaymentDetails` with the following error message via
+`callback.updateWith` if the validation fails.
+
+```json
+'Payment app returned invalid shipping address in response.'
+```
+
+#### Invalid state error message
+
+If Chrome encounters an invalid state upon receiving any of the change requests
+it will call `callback.updateWith` with a redacted `updatePaymentDetails`
+bundle. The bundle will only contain the `error` key with `"Invalid state"`.
+Examples of an invalid state are:
+
+* When Chrome is still waiting for the merchant's response to a previous change
+  (such as an ongoing change event).
+* The payment app provided shipping option identifier does not belong to any of
+  the merchant specified shipping options.
+
+### Receive for the updated payment details from the merchant
+
+```kotlin
+private fun unbind() {
+    if (isBound) {
+        unbindService(mConnection)
+        isBound = false
+    }
+}
+
+private val mCallback: IPaymentDetailsUpdateServiceCallback =
+    object : IPaymentDetailsUpdateServiceCallback.Stub() {
+        override fun paymentDetailsNotUpdated() {
+            // Payment request details have not changed.
+            unbind()
+        }
+
+        override fun updateWith(updatedPaymentDetails: Bundle) {
+            mUpdatedPaymentDetails = updatedPaymentDetails
+            unbind()
+        }
+    }
+```
+
+`updatePaymentDetails` is the bundle equivalent to the
+[`PaymentRequestDetailsUpdate`](https://w3c.github.io/payment-handler/#the-paymentrequestdetailsupdate)
+WebIDL dictionary (after redacting the `modifiers` field) and contains the
```

Comment:
Link to explanation of WebIDL?

Reply:
https://www.w3.org/TR/WebIDL-1/
I will add a link.